### PR TITLE
Implement node connectivity

### PR DESCRIPTION
**Pull request by AKoll**
First janky implementation of user-created node edges.
Does not work at all on Firefox.

### CODE REVIEW

File: src/components/Canvas.vue [hunk 1]
```diff
@@ -63,22 +76,67 @@ export default {
...
+        event.instance.connecting = true
+        if (this.$refs.newEdge) this.$refs.newEdge.refresh()
+      }
     },
-    addNode: function (component, spec) {
+    onNodeMove (nodeId) {
+      for (let edgeId in this.$refs) {
+        if (edgeId.split('.').includes(nodeId)) {
+          for (let edge of this.$refs[edgeId]) {
+            edge.refresh()
```
Comment (Pikaju) [inline src/components/Canvas.vue]: Ich wein gleich

Comment (Pikaju) [reply src/components/Canvas.vue]: Should this perhaps also refresh newEdge? Vinodh's multi-touch might lead to visual artifacts, right?

File: src/components/Canvas.vue [hunk 2]
```diff
@@ -63,22 +76,67 @@ export default {
...
+      // Prevent connecting a node to itself
+      if (edge.start.nodeId === edge.end.nodeId) return false
+
+      return true
+    },
+    addEdge (edge) {
+      if (this.isValidEdge(edge)) {
+        this.edges.push(edge)
+        edge.start.connected += 1
+        edge.end.connected += 1
```
Comment (AKoll) [inline src/components/Canvas.vue]: Would a ++ operator be nicer here?

File: src/components/Edge.vue [hunk 3]
```diff
@@ -87,6 +62,40 @@ export default {
...
+      let y = this.end.y
+      if (this.end instanceof HTMLElement) {
+        let rect = this.end.getBoundingClientRect()
+        let parentRect = this.$el.parentElement.getBoundingClientRect()
+        x = rect.x + rect.width / 2 - parentRect.x
+        y = rect.y + rect.height / 2 - parentRect.y
+      }
+      return { x, y }
+    },
+    refresh () {
```
Comment (Pikaju) [inline src/components/Edge.vue]: Why not use `this.$forceUpdate()` instead?

File: src/components/Canvas.vue (hunk 2 above)
Comment (Pikaju) [reply src/components/Canvas.vue]: Yes

File: src/components/Canvas.vue (hunk 1 above)
Comment (AKoll) [reply src/components/Canvas.vue]: Very good point

File: src/components/Edge.vue (hunk 3 above)
Comment (AKoll) [reply src/components/Edge.vue]: It seems that $forceUpdate will not recompute computed values and methods. A custom refresh method allows for more control and optimization. However, I find this unpretty myself.
Might there be an observable way to determine the bounding box or just the center point as an object field?

File: src/components/Canvas.vue [hunk 4]
```diff
@@ -57,28 +69,98 @@ export default {
...
+        if (!this.preserveStart) {
+          if (this.newEdge && event.connector !== this.newEdge.start) {
+            this.newEdge.start = event.isOutput ? event.instance : this.newEdge.start
+            this.newEdge.end = event.isOutput ? this.newEdge.end : event.instance
+            this.addEdge(this.newEdge)
+          }
+          event.instance.connecting = false
+        } else {
+          // TODO: This check is REALLY bad. Needs rework for detecting pulloff on preserveStart
+          if (this.newEdge.end.x > 0 && this.newEdge.end.y > 0) {
```
Comment (AKoll) [inline src/components/Canvas.vue]: This needs a rework

**Comment by Pikaju** [comment]
Mouse cursor looks weird, likely because the canvas itself is not a valid place to end the drag event and so Chrome tries to provide visual feedback.

File: src/components/Canvas.vue [hunk 5]
```diff
@@ -57,28 +69,98 @@ export default {
...
+            this.newEdge.start = this.newEdge.start
```
Comment (ProtoNXT) [inline src/components/Canvas.vue]: This line of code does not do anything and can be removed.

**Comment by AKoll** [comment]
Deprecated by #18

**Pull request closed without merging**